packages/driver/src/cypress/location.coffee

```diff
@@ -141,8 +142,19 @@ class $Location
     ## aws.amazon.com/bucket/foo
     ## foo.bar.co.uk
     ## foo.bar.co.uk/asdf
+    ##url = "/#{url}"
     url = url.split("/")[0].split(".")
-    url.length is 3 or url.length is 4
+
+    ## In the case of a url like /?foo=..
+    ## It's split into ?foo=, "", "" which is not a valid
+    ## So if after we split a url, there are any empty strings consider it invalid
+    ## if these checks would pass that would be great :|
+    ## https://github.com/cypress-io/cypress/issues/5090
+    containEmptyString = url.some (part) -> part is ""
+    if not containEmptyString
+      url.length is 3 or url.length is 4 
+    else 
+      false
```

Comment:
avoid explicit if/else where possible:

```
if containEmptyString
	return false

return url.length is 3 or url.length is 4
```

Reply:
Right, yeah. That was me trying to deal with coffescript as I was seeing weird stuff with returns. But I'll try that.

Reply:
I don't know if you use VS Code, but if you do, try this plugin out: https://marketplace.visualstudio.com/items?itemName=jhessin.coffee2preview

In a .coffee file, you can select a block of code and run the "coffeescript check" command to see what it transpiles to in .js. Useful for when you're not sure what a block of coffeescript will do.

Reply:
Oh nice. I do use vscode. That will definitely come in handy then